src/app/Library/CrudPanel/Traits/Read.php

```diff
@@ -55,8 +55,10 @@ public function getCurrentEntry()
      */
     public function getEntry($id)
     {
+        // build the model query from the main crud query
+        $this->model = $this->model->setQuery($this->query->getQuery())->getModel();
```

Comment:
Ding ding ding! Duplicated code found 😅

![dda2151f8613bfb4698b615b6b3c801d](https://user-images.githubusercontent.com/1032474/162433770-d6b19d71-e759-4fd7-8c5f-d3a4c7bab54f.gif)

Also, previously this method did one thing, GET something. Now... it SETS something and GETS something. So it does two things, and more than its name says. Spidey sense going haywire here 😅

How about we move this `setQuery()` thing to a different method, `getModel()`? Would that solve em? Then inside these methods we could use `$this->getModel()` instead of `$this->model`, which itself takes into account any query you've already set.

Would that make sense?

Reply:
Hahaha I love your gifs! 

Sure, totally agree, just refactored, thanks 🙏